contracts/strategies/qv-hackathon/HackathonQVStrategy.sol

```diff
@@ -0,0 +1,269 @@
+    }
+
+    /// @dev Initializes the strategy.
+    function __HackathonQVStrategy_init(
+        address _easContractAddress,
+        address _schemaRegistryAddress,
+        string memory _schema,
+        bool _revocable,
+        uint256 _poolId,
+        bytes memory _data
+    ) public {
+        (
+            registryGating,
+            metadataRequired,
+            maxVoiceCreditsPerAllocator,
+            registrationStartTime,
+            registrationEndTime,
+            allocationStartTime,
+            allocationEndTime
+        ) = abi.decode(_data, (bool, bool, uint256, uint256, uint256, uint256, uint256));
+        __QVSimpleStrategy_init(
+            _poolId,
+            registryGating,
+            metadataRequired,
+            maxVoiceCreditsPerAllocator,
+            registrationStartTime,
+            registrationEndTime,
+            allocationStartTime,
+            allocationEndTime
+        );
+
+        eas = IEAS(_easContractAddress);
+        schemaRegistry = ISchemaRegistry(_schemaRegistryAddress);
+
+        // Register the schema with the SchemaRegistry contract.
+        // todo: setup a default schema they can use for now.
+        // https://optimism-goerli.easscan.org/schema/create
+        // https://docs.attest.sh/docs/tutorials/create-a-schema
+        // https://github.com/ethereum-attestation-service/eas-contracts/blob/master/contracts/
+
+        // todo: we will want to make a default one and then let the user override it if they want.
+        // _registerSchema(_schema, ISchemaResolver(address(this)), _revocable);
+    }
+
+    /// =========================
+    /// == Internal Functions ===
+    /// =========================
+
+    /// @dev Registers a recipient with the EAS contract.
+    /// @param _data The data to register the recipient with.
+    /// @param _sender The address of the sender.
+    function _registerRecipient(bytes memory _data, address _sender)
+        internal
+        override
+        onlyPoolManager(_sender)
+        returns (address recipientId)
+    {
+        // decode _data
+        // Note: this _data will also contain another data bytes array to pass to _registerRecipient()
+        (address recipient, uint256 amount, address token, bytes memory data) =
+            abi.decode(_data, (address, uint256, address, bytes));
+
+        // check if already added
+        if (hackers.contains(_sender)) {
+            revert ALREADY_ADDED();
+        }
+        // attest the recipient
+        // todo: update the following values to be passed in
+        uint64 expirationTime = uint64(block.timestamp + 100 days);
+        bool revocable = true;
+
+        // create the request
+        AttestationRequest memory attestationRequest = AttestationRequest(
+            // todo: add schema
+            "",
+            AttestationRequestData({
+                recipient: recipient,
```

Comment:
the `registryGating` implementation is missing
it's not part of the spec, but if the pool relays on the registry, we should ensure that the anchor of an identity is used to attest to